package pb;

enum ParamType {
    Nope        = 0;
    Value       = 1;
    OnOff       = 2;
    Boolean     = 3;
    MinMaxStep  = 4;
    Enumeration = 5;
    Button      = 6;
}


message PropertyType {
    required string name          = 1;
    required ParamType type       = 2;

    // MinMaxStep
    optional int32 min            = 3;
    optional int32 max            = 4;
    optional int32 step           = 5;

    // Enumaration
    optional string values        = 6;
    optional string ui_values     = 7;

    // Value
    optional string default_value = 8;

    optional string group         = 9;
    optional string sub_group     = 10;
    optional string description   = 11;

    optional bool ui              = 12;
    optional bool readonly        = 13;

    optional string netHelper     = 14;
}

message ResourceType {
    required int32 id                   = 1;
    required string name                = 2;
    optional string manufacture         = 3;
    repeated int32 parentId             = 4;
    repeated PropertyType propertyType  = 5;
}

message ResourceTypes {
    repeated ResourceType resourceType = 1;
}
